### PR TITLE
[MENFORCER-346] - RequireSnapshotVersion: Fix parent with CI Friendly Version

### PR DESCRIPTION
**Pull request by famod**
Fixes unresolved `${revision}` parent version.

I am not sure whether this is the best approach.
`project.getParentArtifact()` not providing the resolved version might also be considered a Maven core bug.

Following this checklist to help us incorporate your
contribution quickly and easily:
* [x]  Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MENFORCER) filed
  for the change (usually before you start working on it).  Trivial changes like typos do not
  require a JIRA issue.  Your pull request should address just this issue, without
  pulling in other changes.

* [x]  Each commit in the pull request should have a meaningful subject line and body.

* [x]  Format the pull request title like `[MENFORCER-XXX] - Fixes bug in ApproximateQuantiles`,
  where you replace `MENFORCER-XXX` with the appropriate JIRA issue. Best practice
  is to use the JIRA issue title in the pull request title and in the first line of the
  commit message.

* [x]  Write a pull request description that is detailed enough to understand what the pull request does, how, and why.

* [x]  Run `mvn clean verify` to make sure basic checks pass. A more thorough check will
  be performed on your pull request automatically.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

* [x]  I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

### CODE REVIEW

**Comment by famod** [comment]
FYI: a few other rules are also affected by `project.getParentArtifact()` not providing the resolved version.

File: maven-enforcer-plugin/src/it/projects/require-snapshot-version_failure_parent-ci-friendly/pom.xml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+<project>
```
Comment (elharo) [inline maven-enforcer-plugin/src/it/projects/require-snapshot-version_failure_parent-ci-friendly/pom.xml:22]: two space indents for XML files per project style

**Comment by famod** [comment]
> Thanks. This looks significant enough to require an ICLA.

I've just submitted it.

File: maven-enforcer-plugin/src/it/projects/require-snapshot-version_failure_parent-ci-friendly/pom.xml (hunk 1 above)
Comment (famod) [reply maven-enforcer-plugin/src/it/projects/require-snapshot-version_failure_parent-ci-friendly/pom.xml:22]: Not sure I'm following: This is the root element so there should be not indentation here?

Comment (elharo) [reply maven-enforcer-plugin/src/it/projects/require-snapshot-version_failure_parent-ci-friendly/pom.xml:22]: I must have been confused.

**Comment by jira-importer** [comment]
Resolve #459

**Comment by jira-importer** [comment]
Resolve #459

**Pull request merged**